Mutagen.Bethesda.Analyzers.Skyrim/Record/Location/CircularLocationAnalyzer.cs

```diff
@@ -0,0 +1,49 @@
+using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Plugins.Cache;
+using Mutagen.Bethesda.Skyrim;
+
+namespace Mutagen.Bethesda.Analyzers.Skyrim.Record.Location;
+
+class CircularLocationAnalyzer : IContextualRecordAnalyzer<ILocationGetter>
+{
+    public static readonly TopicDefinition CircularLocation = MutagenTopicBuilder.DevelopmentTopic(
+            "Circular Location",
+            Severity.CTD)
+        .WithoutFormatting("Location is its own parent location - this leads to CTDs and infinite loading screens");
+
+    IEnumerable<TopicDefinition> IAnalyzer.Topics { get; } = [CircularLocation];
+
+    private static bool FindCicularLocation(ILocationGetter sourceLocation, ILocationGetter? currentLocation, ILinkCache linkCache)
+    {
+        if (currentLocation == null) return false;
+
+        if (currentLocation.FormKey == sourceLocation.FormKey) return true;
+
+        if (currentLocation.ParentLocation.IsNull) return false;
+
+        if (!linkCache.TryResolve<ILocationGetter>(currentLocation.ParentLocation.FormKey, out var parentLocation)) return false;
+
+        return FindCicularLocation(sourceLocation, parentLocation, linkCache);
+    }
+
+    void IContextualRecordAnalyzer<ILocationGetter>.AnalyzeRecord(ContextualRecordAnalyzerParams<ILocationGetter> param)
+    {
+        var location = param.Record;
+
+        if (location.ParentLocation.IsNull) return;
+
+        if (!param.LinkCache.TryResolve<ILocationGetter>(location.ParentLocation.FormKey, out var parentLocation)) return;
+
+        if (FindCicularLocation(location, parentLocation, param.LinkCache))
+        {
+            param.AddTopic(CircularLocation.Format());
```

Comment:
This should include a note to what sequence of locations leads to this circular parent location to easily fix this. Probably not in the main string, but in the meta data this should best be included. Note that we can pass lists of things into the meta data, but it should not just be a unique string for each location with something like the editor or form key, but better the raw data that can also be used programmatically later, like the form link.